src/monitor/commands/end-monitoring.command.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/database/prisma.service';
+import { MonitorStatus } from '../utils/monitor.enum';
+import { InvalidMonitoringStatusException, MonitoringNotFoundException, NotTheResponsibleProfessorException } from '../utils/exceptions';
+
+@Injectable()
+export class EndMonitoringCommand {
+  constructor(private prisma: PrismaService) {}
+
+  async execute(id_monitoring: number, userId: number): Promise<void> {
+    const AMT_OFFSET_IN_MS = -4 * 60 * 60 * 1000; 
+    const now = new Date();
+    const nowAMT = new Date(now.getTime() + AMT_OFFSET_IN_MS).toISOString();
+
+    const monitoring = await this.prisma.monitor.findUnique({
+      include: {
+        status: true,
+      },
+      where: { id: id_monitoring },
+    })
+  
+    if (!monitoring) {
+      throw new MonitoringNotFoundException();
+    }
+
+    if (userId != monitoring.responsible_professor_id) {
+      throw new NotTheResponsibleProfessorException();
+    }
```

Comment:
O coordenador pode finalizar qualquer monitoria, então isso aqui não tá certo. Se o usuário logado for professor você checa isso, mas se for um coordenador não precisa ser o responsável pelo monitor.

Reply:
Agora que vi que esqueci de botar no card mas é importante. Pode add isso, pf? Alterei o card